### PR TITLE
Use estimatesmartfee for bitcoin tx fee estimation

### PR DESCRIPTION
**Pull request by harshjv**

### CODE REVIEW

File: src/providers/bitcoin/BitcoinLedgerProvider.js [hunk 1]
```diff
@@ -110,14 +110,17 @@ export default class BitcoinLedgerProvider extends LedgerProvider {
...
+  async getUtxosForAmount (amount) {
     const utxosToUse = []
     let addressIndex = 0
     let currentAmount = 0
     let numOutputsOffset = 0
 
     while ((currentAmount < amount)) {
-      const address = await this.getAddressFromIndex(addressIndex)
+      const [ feePerByte, address ] = await Promise.all([
+        this.getMethod('getFeePerByte')(2),
```
Comment (matthewjablack) [inline src/providers/bitcoin/BitcoinLedgerProvider.js]: Why 2 blocks and not 1?

File: src/providers/bitcoin/BitcoinRPCProvider.js [hunk 2]
```diff
@@ -3,6 +3,10 @@ import BigNumber from 'bignumber.js'
 import JsonRpcProvider from '../JsonRpcProvider'
 
 export default class BitcoinRPCProvider extends JsonRpcProvider {
+  async getFeePerByte (numberOfBlocks = 2) {
+    return this.jsonrpc('estimatesmartfee', numberOfBlocks).then(({ feerate }) => (feerate * 1e8) / 1024)
```
Comment (matthewjablack) [inline src/providers/bitcoin/BitcoinRPCProvider.js:7]: what's the calculation here?

**Review by matthewjablack — APPROVED** [review]
LGTM 🎉 just a couple comments.

File: src/providers/bitcoin/BitcoinRPCProvider.js (hunk 2 above)
Comment (harshjv) [reply src/providers/bitcoin/BitcoinRPCProvider.js:7]: `feerate` is in `BTC/kB`. While we are using `sat/b` in our calculations. So this is just converting `BTC -> sat` and then `kB -> b`

File: src/providers/bitcoin/BitcoinLedgerProvider.js (hunk 1 above)
Comment (harshjv) [reply src/providers/bitcoin/BitcoinLedgerProvider.js]: It was just a default value i went with. Best was would be accepting user's preferred feerate/number of block from `sendTransaction` and `getUtxosForAmount`. We can look into that.

**Review by monokh — APPROVED** [review]
LGTM 🎉

File: src/providers/bitcoin/BitcoinRPCProvider.js (hunk 2 above)
Comment (monokh) [reply src/providers/bitcoin/BitcoinRPCProvider.js:7]: @harshjv  `feerate * 1e8` may create some floating point precision problems, may not matter that much but if you want a whole number, should probably use `BigNumber.js` for this calculation

**Pull request merged**